src/index.ts

```diff
@@ -254,34 +254,26 @@ function getConvexPolyhedronParameters (object: Object3D): ShapeParameters<Shape
 		);
 	}
 
-	// Compute the 3D convex hull.
-	const hull = new ConvexHull().setFromObject(new Mesh(geometry));
-	const hullFaces = hull.faces;
-	const vertices: number[] = [];
-	const faces: number[][] = [];
-
-	let currentFaceVertex = 0;
-	for (let i = 0; i < hullFaces.length; i++) {
-		const hullFace = hullFaces[ i ];
-		const face: number[] = [];
-		faces.push(face);
-
-		let edge = hullFace.edge;
-		do {
-			const point = edge.head().point;
-			vertices.push(point.x, point.y, point.z);
-			face.push(currentFaceVertex);
-			currentFaceVertex++;
-			edge = edge.next;
-		} while ( edge !== hullFace.edge );
+	// Get geometry points
+	const points: Vector3[] = []; 
+	for (let i = 0; i < geometry.attributes.position.array.length; i += 3) {
+		points.push(
+			new Vector3(
+				geometry.attributes.position.array[i],
+				geometry.attributes.position.array[i + 1],
+				geometry.attributes.position.array[i + 2]
+			)
+		);
 	}
 
-	const verticesTypedArray = new Float32Array(vertices.length);
-	verticesTypedArray.set(vertices);
+	// Get convex hull faces
+	const faces = new ConvexHull().setFromPoints(points).collectFaces();
+
+	const vertices = geometry.attributes.position.array as Float32Array
 
 	return {
 		type: ShapeType.HULL,
-		params: { vertices: verticesTypedArray, faces },
+		params: { vertices, faces },
```

Comment:
Is this supposed to return the vertices on the hull exterior? It seems like it would include all interior vertices as well here, not sure if that is expected.

Reply:
You're right, this would return vertices deleted by `ConvexHull` - I'll update the PR so this gets vertices from the computed hull.